### PR TITLE
IDx user id submodule

### PR DESCRIPTION
**Pull request by rtoscani**
## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
Add IDx User Id. [https://idx.lat/en.html](https://idx.lat/en.html)

Test parameters for validating new user id
```
pbjs.setConfig({
    userSync: {
        userIds: [{
            name: 'idx',
        }]
    }
});
```

Contact email of the adapter’s maintainer: prebid@idx.lat

### CODE REVIEW

File: modules/idxIdSystem.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  decode(value) {
+    const idxVal = value ? utils.isStr(value) ? value : utils.isPlainObject(value) ? value.id : undefined : undefined;
+    return idxVal ? {
+      'idx': idxVal
+    } : undefined;
+  },
+  /**
+   * performs action to obtain id and return a value in the callback's response argument
+   * @function
+   * @param {SubmoduleParams} configParams
```
Comment (smenzer) [inline modules/idxIdSystem.js:45]:
```suggestion
   * @param {SubmoduleConfig} config
```

**Review by smenzer — COMMENTED** [review]
please note that after https://github.com/prebid/Prebid.js/pull/5803, the first config sent into `getId()`, `decode()`, and `extendId()` now includes the entire config object, not just params, so please make sure to account for that in this PR so your code works properly.

**Comment by smenzer** [comment]
this looks good, but before I merge please add a docs PR to the userId dev docs in https://github.com/prebid/prebid.github.io and provide a link here. thanks!

**Comment by rtoscani** [comment]
Thanks Scott. PR is on the way on docs repo.

**Review by smenzer — APPROVED** [review]
LGTM

**Comment by bretg** [comment]
docs PR https://github.com/prebid/prebid.github.io/pull/2408

**Pull request merged**